### PR TITLE
Update preprocessing_layers guide with tensorflow 2.5 changes

### PR DESCRIPTION
**Pull request by mattdangerw**

### CODE REVIEW

File: guides/preprocessing_layers.py [hunk 1]
```diff
@@ -32,13 +32,15 @@
...
+or count dense representations.
 - `Hashing` layer: performs categorical feature hashing, also known as the "hashing
 trick".
 - `Discretization` layer: turns continuous numerical features into integer categorical
 features.
-- `StringLookup` layer: turns string categorical values into integers indices.
-- `IntegerLookup` layer: turns integer categorical values into integers indices.
+- `StringLookup` layer: turns string categorical values an encoded representation that can be
+read by an `Embedding` layer or `Dense` layer.
+- `IntegerLookup` layer: turns integer categorical values an encoded representation that can be
```
Comment (8bitmp3) [inline guides/preprocessing_layers.py:42]: Suggestion: "turns ... values _into_ an encoded representation"

```suggestion
- `IntegerLookup` layer: turns integer categorical values into an encoded representation that can be
```

File: guides/preprocessing_layers.py [hunk 2]
```diff
@@ -74,10 +76,9 @@
 a sample of the training data. The list of stateful preprocessing layers is:
 
 - `TextVectorization`: holds a mapping between string tokens and integer indices
-- `Normalization`: holds the mean and standard deviation of the features
-- `StringLookup` and `IntegerLookup`: hold a mapping between input values and output
+- `StringLookup` and `IntegerLookup`: hold a mapping between input values and integer
 indices.
-- `CategoryEncoding`: holds an index of input values.
+- `Normalization`: holds the mean and standard deviation of the features
```
Comment (8bitmp3) [inline guides/preprocessing_layers.py:81]: Nit: add a full stop

```suggestion
- `Normalization`: holds the mean and standard deviation of the features.
```

File: guides/preprocessing_layers.py [hunk 3]
```diff
@@ -294,7 +291,7 @@
 seen during `adapt()`). You can configure this by using the `mask_token` and `oov_token`
 constructor arguments  of `StringLookup`.
 
-You can see the `StringLookup` and `CategoryEncoding` layers in action in the example
+You can see the `StringLookup` in action in the example
 [structured data classification from scratch](https://keras.io/examples/structured_data/structured_data_classification_from_scratch/).
```
Comment (8bitmp3) [inline guides/preprocessing_layers.py:295]: Suggestion:

```suggestion
You can see the `StringLookup` in action in the
[Structured data classification from scratch](https://keras.io/examples/structured_data/structured_data_classification_from_scratch/)
example.
```

(Using sentence case)

File: guides/preprocessing_layers.py [hunk 4]
```diff
@@ -273,19 +274,15 @@
 """
 
 # Define some toy data
-data = tf.constant(["a", "b", "c", "b", "c", "a"])
-
-# Use StringLookup to build an index of the feature values
-indexer = preprocessing.StringLookup()
-indexer.adapt(data)
+data = tf.constant([["a"], ["b"], ["c"], ["b"], ["c"], ["a"]])
```
Comment (fchollet) [inline guides/preprocessing_layers.py:277]: Can you remind me why this change is necessary? What was the previous behavior for `data = tf.constant(["a", "b", "c", "b", "c", "a"])` and what is the new behavior?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: guides/preprocessing_layers.py (hunk 4 above)
Comment (mattdangerw) [reply guides/preprocessing_layers.py:277]: It is not actually needed for the adapt data actually, but it is needed for the call time data to one-hot encode output (this is our inconsistent rank 1 behavior between CategoryEncoding and StringLookup issue).

I figured it would be simpler to keep all the inputs rank 2 for now. This goes away when we add "one_hot", we can go back to simple rank one inputs.

**Review by fchollet — APPROVED** [review]
Thanks, please add the autogenerated files.

**Comment by mattdangerw** [comment]
Done!
